js-test/sensor_spec.js

```diff
@@ -0,0 +1,86 @@
+import { expect } from 'chai';
+import { TemperatureHumiditySensor, Location, EmptyLocation } from '../js/temperature_humidity_sensor.js'
+
+const sensor_data = [
+    { "time": "2020-12-24 13:49:16-0500", "type": "F007TH", "channel": "5", "rolling_code": 174, "name": "Music room", "last": 2, "temperature": 19.9, "humidity": 45, "battery_ok": true, "t_hist": 6, "h_hist": 5 },
+    { "time": "2020-12-24 13:22:16-0500", "type": "F007TH", "channel": "5", "rolling_code": 92, "name": "Front porch", "last": 2, "temperature": 39.9, "humidity": 48, "battery_ok": true, "t_hist": 6, "h_hist": 5 }
+]
+
+describe('TemperatureHumiditySensor tests', () => {
+    it('has raw data', (done) => {
+        const sensor = new TemperatureHumiditySensor(sensor_data)
+        expect(sensor.data).to.equal(sensor_data);
+        done();
+    })
+
+    it("has an inside location of Music room", (done) => {
+        const sensor = new TemperatureHumiditySensor(sensor_data)
+        expect(sensor.inside instanceof Location).to.be.true
+        expect(sensor.inside.name).to.equal("Music room")
+        done();
+    });
+
+    it("has an outside location of Front porch", (done) => {
+        const sensor = new TemperatureHumiditySensor(sensor_data)
+        expect(sensor.outside instanceof Location).to.be.true
+        expect(sensor.outside.name).to.equal("Front porch")
+        done();
+    });
+
+    it('returns and EmptyLocation for inside when data is empty', (done) => {
```

Comment:
"returns an EmptyLocation" here and below

Reply:
I wonder though in all these tests if we need to be checking for the specific class or just want to verify that we can call the various methods on it and get the results we expect?

Reply:
Yeah I think we can. If it returns an empty string for the name then that would imply an EmptyLocation.